### PR TITLE
Don't generate bindings for deleted member functions

### PR DESCRIPTION
**Pull request by martinboehme**
See https://github.com/rust-lang/rust-bindgen/issues/2043 for details.

### CODE REVIEW

File: src/ir/function.rs
Comment (emilio) [inline src/ir/function.rs:602]: It might be worth to shuffle the logic a bit to do less work in the common case, something like:

```rust
if cursor.is_inlined_function() {
    if !context.options().generate_inline_functions {
        return Err(..);
    }
    if cursor.is_deleted_function() {
        return Err(..);
    }
}
```

**Review by emilio — APPROVED** [review]
Yeah, this looks sensible to me, thanks! I can't merge right away because github now requires me to approve CI for new contributors, but if it's green I'm happy to merge as is. Though I have a minor suggestion to maybe do a bit less work in the common case.

**Comment by emilio** [comment]
CI looks green except for rustfmt, so will merge with that fixed and my nit addressed. Thanks!

**Comment by martinboehme** [comment]
Awesome! Was about to push fixes, but you were faster than me. Thanks for fixing those two nits for me!

**Pull request closed without merging**
